currency-rates/cmd/main.go

```diff
@@ -23,12 +24,18 @@ func main() {
 
 	l, _ := config.Build()
 	defer l.Sync() // nolint:errcheck
-	logger := l.Sugar()
+	sugaredLogger := l.Sugar()
+
+	appLogger := logger.NewZapLogger(sugaredLogger)
 
 	ctx := context.Background()
 
-	application := app.NewApp(cfg, logger)
-	if err := application.Run(ctx); err != nil {
-		logger.Error(err)
+	application, err := app.NewApp(cfg, appLogger, ctx)
+	if err != nil {
+		appLogger.Error(err)
```

Comment:
Here is better use paniic or Fatal, case when u just logging error, your project will still return os.Exit(0) normal status code.

Reply:
Good point) Done